File: 删除排序链表中的重复元素.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* deleteDuplicates(struct ListNode* head){
    struct ListNode *p,*q;
    p=head;
    if(p==NULL)
        return 0;
    else
    {
        while(p->next!=NULL){
        if(p->next->val==p->val){
            q=p->next;
            p->next=q->next;
            free(q);
        }
        else
            p=p->next;  
    }
    return head;
    }
}

File: 2020年11月/删除排序链表中的重复元素.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head==None or head.next==None:
            return head
        p=head
        q=head.next
        while q:
            if p.val==q.val:
                p.next=q.next
                q=q.next
            else:
                p=q
                q=q.next
        return head